include/pybind11/embed.h

```diff
@@ -198,6 +198,10 @@ inline void finalize_interpreter() {
     if (builtins.contains(id) && isinstance<capsule>(builtins[id])) {
         internals_ptr_ptr = capsule(builtins[id]);
     }
+    // Local internals contains data managed by the current interpreter, so we must clear them to
+    // avoid undefined behaviors when initializing another interpreter
+    detail::get_local_internals().registered_types_cpp.clear();
+    detail::get_local_internals().registered_exception_translators.clear();
```

Comment:
Just to note: I verified that new `TEST_CASE` covers  `.registered_types_cpp.clear()`, but it does not cover `.registered_exception_translators.clear()`.
I think that's fine though, although if someone could add that test coverage in a follow-on PR that would be ideal.

** "covers" in the sense that the test breaks if the call is removed.